component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -411,13 +411,25 @@ public String getName() {
 	 * @param context AuthenticationContext
 	 * @return true, if token is generated successfully
 	 */
-	private boolean generateTOTPToken(AuthenticationContext context) {
-        String username;
-        if (context.getProperty("username") == null) {
+	private boolean generateOTPAndSendByEmail(AuthenticationContext context) {
+		String username = getUsernameFromContext(context);
+
+		if (!TOTPUtil.isSendVerificationCodeByEmailEnabled()) {
+			String appName = context.getServiceProviderName();
+			String tenantDomain = context.getTenantDomain();
+			String sessionDataKey = context.getContextIdentifier();
+
+			String msg = "An attempt was made to send a verification code by email for user: %s for application: %s " +
+					"of %s tenant using sessionDataKey: %s. But sending verification code by email is disabled by " +
+					"admin.";
+			log.warn(String.format(msg, username, appName, tenantDomain, sessionDataKey));
+			return false;
+		}
+
+		if (username == null) {
```

Comment:
better to use StringUtils

Reply:
Addressed with https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/117